numpy/lib/_arraysetops_impl.py

```diff
@@ -335,21 +341,40 @@ def reshape_uniq(uniq):
     output = _unique1d(consolidated, return_index,
                        return_inverse, return_counts,
                        equal_nan=equal_nan, inverse_shape=inverse_shape,
-                       axis=axis)
+                       axis=axis, sorted=sorted)
     output = (reshape_uniq(output[0]),) + output[1:]
     return _unpack_tuple(output)
 
 
 def _unique1d(ar, return_index=False, return_inverse=False,
               return_counts=False, *, equal_nan=True, inverse_shape=None,
-              axis=None):
+              axis=None, sorted=True):
     """
     Find the unique elements of an array, ignoring shape.
+
+    Uses a hash table to find the unique elements if possible.
     """
     ar = np.asanyarray(ar).flatten()
 
     optional_indices = return_index or return_inverse
 
+    if (optional_indices or return_counts) and not sorted:
+        raise ValueError(
+            "`sorted` can only be False if `return_index`, `return_inverse`, "
+            "and `return_counts` are all False."
+        )
+
+    # masked arrays are not supported yet.
+    if not optional_indices and not return_counts and not np.ma.is_masked(ar):
+        # from numpy._core._multiarray_umath import unique_hash
+        from numpy._core._unique import unique_hash
```

Comment:
@ngoldbaum switching the commented status of these two lines to import from `_multiarray_umath` gives a segfault at this line:

https://github.com/numpy/numpy/pull/26018/files#diff-32d1424a27bdc13b58b84b0f68013d5ea1fc5bcf48ad954066b437431e3d5939R138

Not sure why, I could use some hints about this one.